### DISCUSSION TITLE
Reusing ID

### DISCUSSION THREAD

**Discussion by enryfantasy** [rustdesk/rustdesk-server#298] [Q&A]
Hi
I use to set the ID of every client by manually updating Rustdesk.toml id param.
Everything works well unless I reuse an old id.
I dismissed a client and I went to delete the record from peer db table where id='myid'.
Triyng to set the id='myid' into a different client always gives me an error "UUID_MISMATCH":
WARN [src/rendezvous_server.rs:348] Peer myid uuid mismatch: ....

How can I reuse ids? Should I delete something else?

**Comment by enryfantasy** [comment]
Any idea how to solve this. It's a real annoying problem.

**Reply by dinger1986** [reply]
Pro or oss?

**Reply by enryfantasy** [reply]
I'm using the open source version.

**Comment by RipperFox** [comment]
@enryfantasy You can edit the SQLite3 database "db_v2.sqlite3" on the server side to delete complete peers or change ids with simple SQL queries.

**Reply by enryfantasy** [reply]
@RipperFox  I did it. I deleted the entry from the db table "peer" in SQLite (delete from peer where id = 'myid') , but it still keep on saying same error "Peer myid uuid mismatch: ...."
IS it possible that something is written somewhere on disk, outside SQLite? Or maybe there is something on the client?

**Reply by Sven1403** [reply]
Quite late reply but im deleting the whole sqlite3 database file every few weeks/month and reinstall RustDesk then in the company (combined with a update mostly). Works very good.

**Reply by enryfantasy** [reply]
Thanks. I did that a few times, but not as a usual behaviour. I avoided those problem trying to not reuse the same IDs when I replace PCs. Anyway I'll think on that. A good way to clean things up.

**Reply by Sven1403** [reply]
We want to have the computer name (hostname) as ID and when a computer is reinstalled the ID is blocked because it was used before :/ You think its possible to allow duplicates in the sqlite database?

**Reply by enryfantasy** [reply]
Exactly the same problema I face. Hostname as ID. Sometime it works and allow to change PC reusing same hostname and therefore same id, sometime not. I still can't understand why...so by now I change the hostname when I change the computer, to avoid that problem...but it's not the best.

**Reply by Sven1403** [reply]
For me its not a sometimes problem. When a PC gets installed with RustDesk + ChangeID Powershell Script gets called with software deployment, the ID is then blocked completely. You can check this: Open RustDesk -> Call the ChangeID script and you can see that RustDesk is trying to change the ID to the hostname but the name is blocked. RustDesk then generate a new ID. Fun fact: Its case sensetive. So when you have hostname like PC220 and want to use it again its not working. But Pc220 would work :D
